extensions/notebook/src/book/bookTreeView.ts

```diff
@@ -106,27 +110,61 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
 		return books;
 	}
 
+	private getPreviousUri(sec: any[], i: number, root: string): string {
+		// TODO: check for url
+		return i - 1 === -1 ? null : path.join(root, 'content', sec[i - 1].url.concat('.ipynb'));
+	}
+
+	private getNextUri(sec: any[], i: number, root: string): string {
+		// TODO: check for url
+		return i + 1 === sec.length ? null : path.join(root, 'content', sec[i + 1].url.concat('.ipynb'));
+	}
+
 	private getSections(sec: any[], root: string): BookTreeItem[] {
 		let notebooks: BookTreeItem[] = [];
 		for (let i = 0; i < sec.length; i++) {
 			if (sec[i].url) {
 				if (sec[i].external) {
-					let externalLink = new BookTreeItem(sec[i].title, root, sec[i].sections, sec[i].sections ? vscode.TreeItemCollapsibleState.Expanded : vscode.TreeItemCollapsibleState.None, sec[i].url, { command: 'bookTreeView.openExternalLink', title: localize('openExternalLinkCommand', 'Open External Link'), arguments: [sec[i].url], });
+					let externalLink = new BookTreeItem(
+						sec[i].title,
+						root,
+						sec[i].sections,
+						sec[i].sections ? vscode.TreeItemCollapsibleState.Expanded : vscode.TreeItemCollapsibleState.None,
+						sec[i].url,
+						null,
+						null,
+						{ command: 'bookTreeView.openExternalLink', title: localize('openExternalLinkCommand', 'Open External Link'), arguments: [sec[i].url], }
+					);
 					notebooks.push(externalLink);
 				} else {
 					let pathToNotebook = path.join(root, 'content', sec[i].url.concat('.ipynb'));
 					let pathToMarkdown = path.join(root, 'content', sec[i].url.concat('.md'));
 					// Note: Currently, if there is an ipynb and a md file with the same name, Jupyter Books only shows the notebook.
 					// Following Jupyter Books behavior for now
 					if (fs.existsSync(pathToNotebook)) {
-						let notebook = new BookTreeItem(sec[i].title, root, sec[i].sections || sec[i].subsections,
+						let notebook = new BookTreeItem(
+							sec[i].title,
+							root,
+							sec[i].sections || sec[i].subsections,
 							(sec[i].sections || sec[i].subsections) && sec[i].expand_sections ? vscode.TreeItemCollapsibleState.Expanded : sec[i].sections || sec[i].subsections ? vscode.TreeItemCollapsibleState.Collapsed : vscode.TreeItemCollapsibleState.None,
-							sec[i].url, { command: 'bookTreeView.openNotebook', title: localize('openNotebookCommand', 'Open Notebook'), arguments: [pathToNotebook], });
+							sec[i].url,
+							this.getPreviousUri(sec, i, root),
```

Comment:
Yep, with all the reused logic here I definitely believe it's worth putting all this into the BookTreeItem so it's common.